docs/_docs/modules.md

```diff
@@ -0,0 +1,64 @@
+
+## Defining a module class
+Module classes are declared with the `@DesignModuleClass` annotation. The annotation takes no arguments and simply annotates the class so that a customizations() extension function will be generated for the class. Within the class, customizations can be declared in the same way that customizations are declared in a `@DesignComponent` function. However, due to a [bug in KSP][1], different annotations to specify customizations need to be used when declaring them within a module class vs a function. Here is a table of what annotation to use in a function vs a class:
+
+| Function              | Class                         |
+| --------------------- | ----------------------------- |
+| @Design               | @DesignProperty               |
+| @DesignVariant        | @DesignVariantProperty        |
+| @DesignContentTypes   | @DesignContentTypesProperty   |
+| @DesignPreviewContent | @DesignPreviewContentProperty |
+
+Note that once this bug has been fixed, all of the `@Design*Property` annotations will be removed in favor of the analogous annotations used for functions.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModule(
+    @DesignProperty(node = "#text") val text: String,
+    @DesignProperty(node = "#replace")
+    val replaceNode: @Composable (ComponentReplacementContext) -> Unit,
+    @DesignVariantProperty(property = "#text-style") val textStyle: TextStyle,
+    @DesignContentTypesProperty(nodes = ["#button", "#header"])
+    @DesignPreviewContentProperty(
+        name = "Buttons",
+        nodes = [PreviewNode(3, "#button")]
+    )
+)
+```
+
+## Nested Modules
+Module classes can include other modules. The resulting set of customizations will contain all customizations from the current class as well as all nested module classes. Use the `@DesignModuleProperty` annotation to achieve this, using the class name directly as the type of the `@DesignModuleProperty` class property.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModuleCombined(
+    @DesignProperty(node = "#name") val name: String,
+    @DesignModuleProperty val textProperties: TextModule,
+)
+```
+
+## Usage in a function
+Module classes by themselves are not useful until used as a parameter in a `@DesignComponent` function. Use the `@DesignModule` annotation for parameters in such a function. The generated function will combine any customizations `@Design` and `@DesignVariant` parameters in the function with all the embedded customizations from parameters that are modules.
+
+### Example:
+```kotlin
+interface ModuleExample {
+    @DesignComponent(node = "#stage")
+    fun Main(
+        @Design(node = "#header") header: String,
+        @DesignModule moduleCustomizations: TextModuleCombined,
```

Comment:
I think we need usage example of this, showing the generated composable and how the user would set values for the customizations.

Reply:
done